buster/busterbot.py

```diff
@@ -3,29 +3,41 @@
 from functools import lru_cache
 
 import numpy as np
-import openai
 import pandas as pd
-from openai.embeddings_utils import cosine_similarity, get_embedding
+from openai.embeddings_utils import get_embedding
 
 from buster.completers import completer_factory
 from buster.completers.base import Completion
 from buster.formatters.documents import document_formatter_factory
 from buster.formatters.prompts import prompt_formatter_factory
 from buster.retriever import Retriever
 from buster.tokenizers import tokenizer_factory
-from buster.validators.base import validator_factory
+from buster.validators.base import Validator, validator_factory
 
 logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
 
 @dataclass(slots=True)
-class Response:
+class BusterAnswer:
     user_input: str
     completion: Completion
+    validator: Validator = None
     error: bool | None = None
+    _documents_relevant: bool | None = None
     matched_documents: pd.DataFrame | None = None
 
+    @property
+    def documents_relevant(self):
+        """Calls the validator to check if sources were used or not."""
+        if self._documents_relevant is None:
+            logger.info("checking for document relevance")
+            self._documents_relevant = self.validator.check_sources_used(self.completion)
+        return self._documents_relevant
+
+    def to_json():
+        pass
```

Comment:
I can write this as part of my feedback PR

Reply:
great, i put a `NotImplementedError` for now